transport-native-io_uring/src/main/java/io/netty/channel/uring/IOUringEventLoop.java

```diff
@@ -0,0 +1,283 @@
+                                                              .recvBufAllocHandle();
+                final ChannelPipeline pipeline = event.getAbstractIOUringChannel().pipeline();
+
+                allocHandle.lastBytesRead(res);
+                if (allocHandle.lastBytesRead() > 0) {
+                    allocHandle.incMessagesRead(1);
+                    try {
+                        pipeline.fireChannelRead(abstractIOUringServerChannel
+                                                         .newChildChannel(allocHandle.lastBytesRead()));
+                    } catch (Exception e) {
+                        e.printStackTrace();
+                    }
+                    allocHandle.readComplete();
+                    pipeline.fireChannelReadComplete();
+                }
+            }
+
+            //Todo refactoring method name
+            event.getAbstractIOUringChannel().executeReadEvent();
+        break;
+        case READ:
+            System.out.println("EventLoop Read Res: " + res);
+            System.out.println("EventLoop Fd: " + event.getAbstractIOUringChannel().getSocket().intValue());
+            ByteBuf byteBuf = event.getReadBuffer();
+            int localReadAmount = res;
+            if (localReadAmount > 0) {
+                byteBuf.writerIndex(byteBuf.writerIndex() + localReadAmount);
+            }
+
+            final IOUringRecvByteAllocatorHandle allocHandle =
+                    (IOUringRecvByteAllocatorHandle) event.getAbstractIOUringChannel().unsafe()
+                                                          .recvBufAllocHandle();
+            final ChannelPipeline pipeline = event.getAbstractIOUringChannel().pipeline();
+
+            allocHandle.lastBytesRead(localReadAmount);
+            if (allocHandle.lastBytesRead() <= 0) {
+                // nothing was read, release the buffer.
+                byteBuf.release();
+                byteBuf = null;
+                break;
+            }
+
+            allocHandle.incMessagesRead(1);
+            //readPending = false;
+            pipeline.fireChannelRead(byteBuf);
+            byteBuf = null;
+            allocHandle.readComplete();
+            pipeline.fireChannelReadComplete();
+            event.getAbstractIOUringChannel().executeReadEvent();
+        break;
+        case WRITE:
+            System.out.println("EventLoop Write Res: " + res);
+            System.out.println("EventLoop Fd: " + event.getAbstractIOUringChannel().getSocket().intValue());
+            System.out.println("EventLoop Pipeline: " + event.getAbstractIOUringChannel().eventLoop());
+            ChannelOutboundBuffer channelOutboundBuffer = event
+                    .getAbstractIOUringChannel().unsafe().outboundBuffer();
+            //remove bytes
+            int localFlushAmount = res;
+            if (localFlushAmount > 0) {
+                channelOutboundBuffer.removeBytes(localFlushAmount);
+            }
```

Comment:
For non-blocking case do we need to `POLLOUT` here on `-EAGAIN` if we aren't assuming FAST_POLL capability?

Reply:
`POLLOUT` it is already planned, is your idea to POLLOUT before WRITEV(link) event?

Reply:
If initial target is 5.7 then IIUC the kernel handles with FAST_POLL and so it's a non-issue. My comment above was w.r.t to pre-5.7 case